airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -407,7 +407,7 @@ def delete_file(
         :type kwargs: object
         """
         if is_prefix:
-            blobs_to_delete = self.get_blobs_list(container_name, prefix=blob_name, **kwargs)
+            blobs_to_delete = self.get_blobs_list(container_name, prefix=blob_name, delimiter='', **kwargs)
```

Comment:
I prefer this delimiter is not hardcoded.

Reply:
Hi @ephraimbuddy, thanks a lot for reviewing. 

Using a delimiter other than the empty string `''` might lead to non-terminal blob _paths_. Would you suggest declaring the delimiter as a local variable, and then passing it to `get_blobs_list` instead?

Reply:
Yes @monti-python , I suggest that the delete_file method takes a delimiter argument with default of ''(empty string)

Reply:
Brilliant, so the user could limit the deletion depth by specifying a custom `delimiter` and `ignore_if_missing = True`. Thanks for that @ephraimbuddy ! PR updated